### PR TITLE
Specify axis parameter for DequantizeLinear when input rank is 1

### LINKED ISSUE #6067: Clarify default value of `axis` for `DequantizeLinear` when input rank is 1
# Ask a Question

### Question
[DequantizeLinear](https://onnx.ai/onnx/operators/onnx__DequantizeLinear.html) says both that
- the default value of `axis` is `1` 
- and that for `axis`, the `accepted range is [-r, r-1] where r = rank(input)`.

What happens when the rank of the input is 1 and `axis` is not specified?
According to the spec, this seems to be illegal.
`onnxruntime` successfully executes such models.

### PR DESCRIPTION
**Pull request by isdanni**
### Description

Close #6067

### CODE REVIEW

**Comment by justinchuby** [comment]
Could you update https://github.com/onnx/onnx/blob/c459890aa266d74ca612f334d0f3e869dcdbb597/onnx/defs/quantization/defs.cc#L167 and https://github.com/onnx/onnx/blob/c459890aa266d74ca612f334d0f3e869dcdbb597/onnx/defs/quantization/old.cc#L113 ? The md files will be auto updated.

File: onnx/defs/quantization/old.cc [hunk 1]
```diff
@@ -110,7 +110,10 @@ ONNX_OPERATOR_SET_SCHEMA(
         .Output(0, "y", "N-D full precision output tensor. It has same shape as input 'x'.", "T2")
         .Attr(
             "axis",
-            "(Optional) The axis of the dequantizing dimension of the input tensor. Ignored for per-tensor quantization. Negative value means counting dimensions from the back. Accepted range is [-r, r-1] where r = rank(input).",
+            "(Optional) The axis of the dequantizing dimension of the input tensor. Used only for per-axis and blocked "
```
Comment (justinchuby) [inline onnx/defs/quantization/old.cc:113]: Blocked quantization is not supported in the old opset: [Blocked quantization](https://github.com/onnx/onnx/blob/c459890aa266d74ca612f334d0f3e869dcdbb597/onnx/version_converter/adapters/q_dq_21_20.h#L63C30-L63C50)

Comment (isdanni) [reply onnx/defs/quantization/old.cc:113]: Thanks so much!! Will update this asap

**Review by gramalingam — APPROVED** [review]

File: onnx/defs/quantization/old.cc (hunk 1 above)
Comment (justinchuby) [reply onnx/defs/quantization/old.cc:113]: Was this updated?

**Review by linkerzhang — APPROVED** [review]

File: onnx/defs/quantization/old.cc (hunk 1 above)
Comment (isdanni) [reply onnx/defs/quantization/old.cc]: Thanks for double checking this! Just removed “blocked quantisation” from old.cc

**Review by justinchuby — APPROVED** [review]

**Pull request merged**